include/onnxruntime/core/framework/float16.h

```diff
@@ -23,23 +23,143 @@ namespace onnxruntime {
 
 // MLFloat16
 struct MLFloat16 {
+  // uint16_t values for some special values
+  enum Constants : uint16_t {
```

Comment:
enum class.
Also any reason this should be public?

Reply:
`enum class` would create an additional scope that I think is not warranted. All our APIs are struct, so everything is public.

Reply:
There are two issues:

1. Making them private prevents from using them when instantiating well known static constants where those constants are used for construction in the data_types.cc
2. I will try `static constexpr uint16_t `, but what is the gain? Enums are constants and do not require space allocation as required by statics.

Reply:
Done